rollout/analysis.go

```diff
@@ -41,44 +41,49 @@ func (c *RolloutController) getAnalysisRunsForRollout(rollout *v1alpha1.Rollout)
 	return ownedByRollout, nil
 }
 
-func (c *RolloutController) reconcileAnalysisRuns(roCtx *canaryContext) error {
+func (c *RolloutController) reconcileAnalysisRuns(roCtx *canaryContext) (bool, error) {
```

Comment:
See my other comment, I think instead of making sure we are returning true/false correctly in this function, I think the caller can check for `roCtx.PauseContext().HasAddPause()` after calling this function. With my suggested approach, it means we don't have to ensure we correctly return true/false everywhere, because there is a single source of truth.

Reply:
Yeah I like that a lot more